In [4]:
import os

gps_path = "/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/gps_with_4/"
file_list = []
for file in os.listdir(gps_path):
    file_list.append(gps_path + file)

len(file_list)

7661

## Save drums into new .gp file and save the rest into a new .gp file

In [11]:
import guitarpro as gp

output_directory = "/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/gps_drums/"

# Parse the Guitar Pro file
gp_file = gp.parse(file_list[666])
_ = gp_file
# print(gp_file)
filename = file.split('/')[-1]

# Filter tracks based on MIDI channel and percussion check
for track in gp_file.tracks:
    # print(track)
    _.tracks = []
    if track.isPercussionTrack:
        print('drums')
        _.tracks.append(track)
        gp.write(_, output_directory + filename)

drums


In [33]:
import os
import shutil
from tqdm import tqdm
import guitarpro as gp

# Directory to save filtered files
output_directory_drums = "/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/gps_drums/"
output_directory_others = '/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/gps_dg-dg-b/'

# Make sure the output directory exists
os.makedirs(output_directory_drums, exist_ok=True)
os.makedirs(output_directory_others, exist_ok=True)

for file in tqdm(file_list):
    try:
        # Parse the Guitar Pro file
        gp_file = gp.parse(file)
        drums = gp_file
        others = gp_file
        filename = file.split('/')[-1]

        # Filter tracks based on MIDI channel and percussion check
        for track in gp_file.tracks:
            if track.isPercussionTrack:
                drums.tracks = []
                drums.tracks.append(track)
                gp.write(drums, output_directory_drums + filename)
                others.tracks = []
            else:
                others.tracks.append(track)
        
        gp.write(others, output_directory_others + filename)

    except Exception as e:
        print("Could not process {}: {}".format(gp_file, e))



100%|██████████| 7661/7661 [24:10<00:00,  5.28it/s]


In [38]:
file_list_drums = []
file_list_others = []
for file in os.listdir('/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/gps_drums'):
    file_list_drums.append(file)

for file in os.listdir('/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/gps_dg-dg-b'):
    file_list_others.append(file)


print(len(file_list_drums))
print(len(file_list_others))

7662
7662


## Assess count of .txt files

In [12]:
import os

path_drums = '/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/txts_drums'
path_others = '/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/txts_dg-dg-b'
txts_drums = []
txts_others = []

for file in os.listdir(path_drums):
    txts_drums.append(file)

for file in os.listdir(path_others):
    txts_others.append(file)

print(len(txts_drums))
print(len(txts_others))

7654
7636


There are 7654 files in 'txts_drums' and 7636 files in 'txts_others'. Some files got lost in conversion.

## Script to even the files in both folders

In [19]:
import os

# Paths to the folders
drums_folder = "/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/txts_drums"
others_folder = "/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/txts_dg-dg-b"

# List files in each folder
drums_files = set(os.listdir(drums_folder))
others_files = set(os.listdir(others_folder))

# Check for missing files
missing_in_others = drums_files - others_files  # Files in drums but not in others
missing_in_drums = others_files - drums_files  # Files in others but not in drums

# Flag files missing in 'txts_others'
if missing_in_others:
    print("Files in 'txts_drums' but missing in 'txts_others':")
    for file in missing_in_others:
        print("- {}".format(file))
else:
    print("No files are missing in 'txts_others'.")

# Flag files missing in 'txts_drums'
if missing_in_drums:
    print("\nFiles in 'txts_others' but missing in 'txts_drums':")
    for file in missing_in_drums:
        print("- {}".format(file))
else:
    print("\nNo files are missing in 'txts_drums'.")

# Summary
print("\nSummary:")
print("Total files missing in 'txts_others': {}".format(len(missing_in_others)))
print("Total files missing in 'txts_drums': {}".format(len(missing_in_drums)))


Files in 'txts_drums' but missing in 'txts_others':
- New Found Glory - Hit Or Miss.gp3.tokens.txt
- Noir DCsir - Comme Elle Vient.gp3.tokens.txt
- Dr Feelgood - Milk And Alcohol.gp3.tokens.txt
- .DS_Store
- Ruiz, Jacques - Alternative Rock Number 1.gp3.tokens.txt
- Malmsteen, Yngwie - Evil Eye.gp3.tokens.txt
- Tequila Baby - Velhas Fotos (2).gp3.tokens.txt
- Iggy Pop - No Fun (2).gp3.tokens.txt
- Rush - Fly By Night.gp3.tokens.txt
- ALMAFUERTE - Hombre Peste.gp3.tokens.txt
- UFO - Rock Bottom.gp3.tokens.txt
- Hermetica - Buscando Razon (2).gp3.tokens.txt
- Hermetica - Soy De La Esquina.gp3.tokens.txt
- Muse - Bliss (live).gp3.tokens.txt
- ALMAFUERTE - Aguante Bonavena.gp3.tokens.txt
- Noir D,sir - Comme Elle Vient.gp3.tokens.txt
- Faith No More - Ricochet.gp3.tokens.txt
- Osbourne, Ozzy - Flying High Again.gp3.tokens.txt

No files are missing in 'txts_drums'.

Summary:
Total files missing in 'txts_others': 18
Total files missing in 'txts_drums': 0


In [20]:
# Remove these files from 'txts_drums'
if missing_in_others:
    print("Removing extra files from 'txts_drums'...")
    for file in missing_in_others:
        file_path = os.path.join(drums_folder, file)
        try:
            os.remove(file_path)
            print("Removed: {}".format(file))
        except Exception as e:
            print("Error removing {}: {}".format(file, e))
else:
    print("No extra files found in 'txts_drums'.")

# Summary
print("Total extra files removed: {}".format(len(missing_in_others)))

Removing extra files from 'txts_drums'...
Removed: New Found Glory - Hit Or Miss.gp3.tokens.txt
Removed: Noir DCsir - Comme Elle Vient.gp3.tokens.txt
Removed: Dr Feelgood - Milk And Alcohol.gp3.tokens.txt
Removed: .DS_Store
Removed: Ruiz, Jacques - Alternative Rock Number 1.gp3.tokens.txt
Removed: Malmsteen, Yngwie - Evil Eye.gp3.tokens.txt
Removed: Tequila Baby - Velhas Fotos (2).gp3.tokens.txt
Removed: Iggy Pop - No Fun (2).gp3.tokens.txt
Removed: Rush - Fly By Night.gp3.tokens.txt
Removed: ALMAFUERTE - Hombre Peste.gp3.tokens.txt
Removed: UFO - Rock Bottom.gp3.tokens.txt
Removed: Hermetica - Buscando Razon (2).gp3.tokens.txt
Removed: Hermetica - Soy De La Esquina.gp3.tokens.txt
Removed: Muse - Bliss (live).gp3.tokens.txt
Removed: ALMAFUERTE - Aguante Bonavena.gp3.tokens.txt
Removed: Noir D,sir - Comme Elle Vient.gp3.tokens.txt
Removed: Faith No More - Ricochet.gp3.tokens.txt
Removed: Osbourne, Ozzy - Flying High Again.gp3.tokens.txt
Total extra files removed: 18


In [21]:
import os

path_drums = '/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/txts_drums'
path_others = '/Users/pedro/Desktop/DadaGP-v1.1/data-prep-acc-gen/txts_dg-dg-b'
txts_drums = []
txts_others = []

for file in os.listdir(path_drums):
    txts_drums.append(file)

for file in os.listdir(path_others):
    txts_others.append(file)

print(len(txts_drums))
print(len(txts_others))

7636
7636


In [23]:
# asserting we have the same files in each
print(set(txts_drums) == set(txts_others))

True
